src/runtime_src/driver/xclng/drm/xocl/subdev/mb_scheduler.c

```diff
@@ -2825,12 +2823,41 @@ static void destroy_client(struct platform_device *pdev, void **priv)
 	client_release_implicit_cus(exec, client);
 	bitmap_zero(client->cu_bitmap, MAX_CUS);
 
-	list_del(&client->link);
 	DRM_INFO("client exits pid(%d)\n", pid);
+}
+
+static void abort_all_clients(struct platform_device *pdev)
+{
+	struct exec_core *exec = platform_get_drvdata(pdev);
+	struct xocl_dev	*xdev = xocl_get_xdev(pdev);
+	struct list_head *ptr;
+	struct client_ctx *entry;
+
+	mutex_lock(&xdev->dev_lock);
+	list_for_each(ptr, &xdev->ctx_list) {
+		entry = list_entry(ptr, struct client_ctx, link);
+		if (!entry->abort)
+			abort_client(pdev, entry);
+	}
+	mutex_unlock(&xdev->dev_lock);
+
+	wake_up_interruptible(&exec->poll_wait_queue);
```

Comment:
Once a client is aborted, what prevents it from coming back again through execbuf/wait?